skin/src/main/java/org/researchstack/skin/ui/fragment/ActivitiesFragment.java

```diff
@@ -139,76 +231,148 @@ public void onDataAuth() {
-                    : R.color.rss_one_time_color);
+        @Override
+        public int getItemCount() {
+            return tasks.size();
+        }
 
-            holder.title.setText(Html.fromHtml("<b>" + task.taskTitle + "</b>"));
-            holder.title.append("\n" + task.taskCompletionTime);
-            holder.title.setTextColor(tintColor);
+        @Override
+        public int getItemViewType(int position) {
+            Object item = tasks.get(position);
+            return item instanceof Header ? VIEW_TYPE_HEADER : VIEW_TYPE_ITEM;
+        }
 
-            Drawable drawable = holder.dailyIndicator.getDrawable();
-            drawable = DrawableCompat.wrap(drawable);
-            DrawableCompat.setTint(drawable, tintColor);
-            holder.dailyIndicator.setImageDrawable(drawable);
+        // Clean all elements of the recycler
+        public void clear() {
+            tasks.clear();
+            notifyDataSetChanged();
+        }
 
-            holder.itemView.setOnClickListener(v -> {
-                LogExt.d(getClass(), "Item clicked: " + task.taskID);
-                publishSubject.onNext(task);
-            });
+        // Add a list of items
+        public void addAll(List<Object> list) {
+            tasks.addAll(list);
+            notifyDataSetChanged();
         }
 
-        @Override
-        public int getItemCount() {
-            return tasks.size();
+        public static class HeaderViewHolder extends RecyclerView.ViewHolder {
+
+            TextView title;
+            TextView message;
+
+            public HeaderViewHolder(View itemView) {
+                super(itemView);
+                title = (TextView) itemView.findViewById(R.id.activity_header_title);
+                message = (TextView) itemView.findViewById(R.id.activity_header_message);
+            }
         }
 
         public static class ViewHolder extends RecyclerView.ViewHolder {
+            View colorBar;
             ImageView dailyIndicator;
-            AppCompatTextView title;
+            TextView title;
+            TextView subtitle;
 
             public ViewHolder(View itemView) {
                 super(itemView);
+                colorBar = itemView.findViewById(R.id.color_bar);
                 dailyIndicator = (ImageView) itemView.findViewById(R.id.daily_indicator);
-                title = (AppCompatTextView) itemView.findViewById(R.id.task_title);
+                title = (TextView) itemView.findViewById(R.id.task_title);
+                subtitle = (TextView) itemView.findViewById(R.id.task_subtitle);
             }
         }
+
+        public static class Header {
+            String title;
+            String message;
+
+            public Header(String t, String m) {
+                title = t;
+                message = m;
+            }
+        }
+
+        // TODO: this should live somewhere else like ResourceManager?  Cause it will be app specific
+        //       and the constants defined somewhere?
```

Comment:
I think it's fine to hard-code certain tasks to colors here for the default behavior, without magic constants, but I think it is important for the user to be able to inject their own ViewHolders easily.  Each app we have done on iOS has had a different Activities UI, so we should focus on making it customizable by the developer.